### PR TITLE
fix(authentication-service): fix keycloak logout after token refresh

### LINKED ISSUE #596: Logout for keycloak does not work after token refresh
**Describe the bug**
Keycloak Logout does not work after refresh token api is hit.

**To Reproduce**
1. Create token using keycloak code
2. Refresh the token once
3. Hit the logout api
4. The user is not logged out of keycloak

**Expected behavior**
The user should have been logged out of keycloak

### PR DESCRIPTION
**Pull request by akshatdubeysf**
## Description

 fix keycloak logout after token refresh

Fixes #596

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## Checklist:

- [X] Performed a self-review of my own code
- [X] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [X] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/authentication-service/src/modules/auth/login.controller.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ import {
 } from 'loopback4-authentication';
 import {authorize, AuthorizeErrorKeys} from 'loopback4-authorization';
 import moment from 'moment-timezone';
+import {ExternalTokens} from '../..';
```
Comment (samarpanB) [inline services/authentication-service/src/modules/auth/login.controller.ts:38]: this sometimes fail. make it '../types'

File: services/authentication-service/src/types.ts [hunk 2]
```diff
@@ -23,3 +23,8 @@ export interface IAuthClientDTO {
   client_secret: string;
   // sonarignore:end
 }
+
+export interface ExternalTokens {
+  externalAuthToken: string;
```
Comment (samarpanB) [inline services/authentication-service/src/types.ts:28]: make these optional

**Review by samarpanB — CHANGES_REQUESTED** [review]

File: services/authentication-service/src/modules/auth/login.controller.ts (hunk 1 above)
Comment (akshatdubeysf) [reply services/authentication-service/src/modules/auth/login.controller.ts]: fixed

File: services/authentication-service/src/types.ts (hunk 2 above)
Comment (akshatdubeysf) [reply services/authentication-service/src/types.ts]: fixed

**Review by samarpanB — APPROVED** [review]

**Pull request merged**